pkg/builder/builder_steps.go

```diff
@@ -140,7 +141,17 @@ func generateProjectSettings(ctx *Context) error {
 
 func injectDependencies(ctx *Context) error {
 	// Add dependencies from build
-	for _, d := range ctx.Build.Dependencies {
+	return InjectDependenciesCommon(&ctx.Maven.Project, ctx.Build.Dependencies, ctx.Catalog)
+}
+
+// InjectDependenciesCommon --
+func InjectDependenciesCommon(
```

Comment:
We should avoid this `*Common` suffix, I'd rather move those functions to a dedicated package or rethink the names used for the steps

Reply:
I think this function belongs in a utils file rather than here. It performs a role that is independent of anything other than a Maven project + Dependencies + Catalog which don't necessarily have to be inside the builder. I've moved the function to a new file under utils/camel. I have also moved the function which sanitizes integration dependencies.